core/src/main/java/org/apache/calcite/rel/core/Sort.java

```diff
@@ -124,11 +125,19 @@ public abstract Sort copy(RelTraitSet traitSet, RelNode newInput,
 
   @Override public @Nullable RelOptCost computeSelfCost(RelOptPlanner planner,
       RelMetadataQuery mq) {
+    double rowCount = mq.getRowCount(this);
+    if (collation.getFieldCollations().isEmpty()) {
+      return planner.getCostFactory().makeCost(rowCount, 0, 0);
+    }
+    final int offsetValue = offset == null ? 0 : RexLiteral.intValue(offset);
```

Comment:
Theoretically, `offset` could not be a `RexLiteral` (it could be a `RexCall` or a `RexDynamicParam`), which would lead to an `AssertionError` when calling `RexLiteral.intValue`. To be on the safe side, this should be checked before calling this function.

Reply:
Thanks, I set offset to be 0 if the it is instance of RexDynamicParam as follow. Am I right ?
     offsetValue = offset == null || offset instanceof RexDynamicParam ? 0 : RexLiteral.intValue(offset);

When offset is a RexCall, the RexLiteral.intValue will compute the final value.